spec/support/thread_order_support.rb

```diff
@@ -0,0 +1,93 @@
+class ThreadOrderSupport
+  def initialize
+    @bodies  = {}
+    @threads = []
+    @queue   = [] # we may not have thread stdlib required, so may not have Queue class
```

Comment:
We care about minimizing stdlibs loaded from RSpec itself, but loading stdlibs in specs to help you test things like this is fine -- so I'd recommend just using `Queue` over `[]` with a mutex.  Stdlibs or other libraries loaded in the specs are isolated from the "minimize loaded stdlibs" library-wide spec I mentioned above because that one shells out and starts a new ruby process, loading all of the lib files of the repo and thus avoiding pollution by spec files loading needed libraries.